src/ui/molecules/stepper/Stepper.tsx

```diff
@@ -177,157 +122,119 @@ export const Stepper: React.FC<StepperProps> = ({
   const { isXSmall, isSmall }: Breakpoints = useBreakpoint()
   const isMobile = (): boolean => isXSmall || isSmall
 
-  const [state, dispatch]: UseReducer<StepperState, StepperAction> = useReducer<
-    Reducer<StepperState, StepperAction>,
-    DeepReadonly<ImmutableList<Step>>
-  >(stepperReducer, ImmutableList(steps), initState)
-
-  const handlePreviousClick = useCallback((): void => {
-    dispatch({ type: 'previousClicked' })
-  }, [])
+  const currentStep = useMemo(() => steps[current], [steps, current])
 
-  const handleNextClick = useCallback((): void => {
-    const clickOnNextSucceed =
-      !steps[state.activeStepIndex].onValidate || steps[state.activeStepIndex].onValidate?.()
-    if (clickOnNextSucceed) {
-      onNext?.()
-      dispatch({ type: 'stepCompleted' })
-    } else {
-      dispatch({ type: 'stepFailed' })
-    }
-  }, [onNext, state.activeStepIndex, steps])
+  const isFirstStep = useMemo(() => current === 0, [current])
 
-  const handleSubmit = useCallback((): void => {
-    const isStepValid =
-      !steps[state.activeStepIndex].onValidate || steps[state.activeStepIndex].onValidate?.()
-    if (isStepValid) {
-      onSubmit?.()
-      dispatch({ type: 'stepperSubmitted' })
-    } else {
-      dispatch({ type: 'stepFailed' })
-    }
-  }, [onSubmit, state.activeStepIndex, steps])
+  const isLastStep = useMemo(() => current === steps.length - 1, [steps, current])
 
-  const handleReset = useCallback((): void => {
-    onReset?.()
-    dispatch({ type: 'stepperReseted', payload: ImmutableList(steps) })
-  }, [onReset, steps])
+  const getStepStatus = (step: DeepReadonly<Step>, index: StepIndex): StepState =>
```

Comment:
This component is dumb, so please move this computation to a dedicated `selector` that will inject the view model as a prop.